### ISSUE TITLE
Internalization files update

### ISSUE THREAD

**Issue by jaumard** [flutter/flutter#19165]
By reading the documentation of Internationalization I have to create manually all the `intl_*lang*.arb` files. But once it's done and I need to update my translation (example I removed and/or added screens), how can I update the localisations ? What's the procedure in that case ? 

Because having to manually search for the new messages into `intl_messages.arb` and after add them into each `intl_*lang*.arb` is not really an option... 

Having the Intl package that merge all the translation and add/remove/update for all langage.
On angular the i18n tool is doing it and I find it really cool to work with.

**Comment by ebeem** [comment]
run this script in your project directory to generate translations from .arb files. You have to specify them all, here I am using ar & en in addition to the default messages.
Do not forget to change your `output-dir` and path of your .arb files if they differ.

`flutter pub pub run intl_translation:generate_from_arb --output-dir=lib/l10n --no-use-deferred-loading lib/localization/localizations.dart lib/l10n/intl_ar.arb lib/l10n/intl_en.arb lib/l10n/intl_messages.arb`

note: you have to add the dependency `intl_translation: ">=0.16.6"` to your `pubspec.yaml` in `dev_dependencies` section to run this script

**Comment by jaumard** [comment]
Thanks @ebeem I have no problem generating all the files, it's working great my concern is more than the update has to be almost full manual... If you have two/tree language it can be manageable but once you have 10/20 language much less doable :)

**Comment by zoechi** [comment]
If you use https://translate.google.com/ it remembers previously made translations.
Then you just have to add the missing ones.

**Comment by guyzk** [comment]
This is really a painful issue and I do hope that the flutter team will make a localization tool instead of this tiring process.

**Comment by edlman** [comment]
I'd like to see something similar to (or same as) `gettext`, `msgmerge` and `msgformat` in Dart/Flutter. Its format is quite simple and easy to use, it's supported on almost all platforms and it supports incremental and fuzzy translations. And there are online and offline (windows, mac, linux) tools for translations. It's a pitty Google didn't choose it for Dart and created own system which lacks some important functionalities.
There is a Dart package https://pub.dev/packages/gettext, which may be a solution. I didn't try it yet.

**Comment by cielo** [comment]
I was curious about this too. If new texts are added to internationalized app, then new fresh `arb` files will need to be generated.

The thing is that if there are hundreds of translations already done for languages, but a dozen new translations need to be added, it is quite troublesome and time-consuming to find and update those texts out of hundreds of texts.

More so, all these steps have to be done manually, and there is no graceful way to merge or incrementally update arb files. Is there a way to easily merge or incrementally update arb files? e.g. Some thing similar to what `git merge` does.

**Comment by aleksakrstic** [comment]
You can use Flutter Intl plugin for IDE to get rid of manual execution of `intl_translation` tool whenever you add new string key to ARB files:

- VS Code: https://marketplace.visualstudio.com/items?itemName=localizely.flutter-intl
- Android Studio: https://plugins.jetbrains.com/plugin/13666-flutter-intl

For translations from main language to remaining languages, you can try [Localizely](https://localizely.com/) platform.

**Comment by cielo** [comment]
> You can use Flutter Intl plugin for IDE to get rid of manual execution of `intl_translation` tool whenever you add new string key to ARB files:

That seems to be a great extension to avoid running commands manually, and maintaining source of truth in `arb` file. 

But.. it does seem to create more work in maintaining `arb` files now. For instance, if I add/update/remove dozens of translations in source English arb file, I will still have to manually do the same thing in dozens of other localization `arb` files, manually. 

Because `arb` files are now have to be maintained manually, I think it can get more error prone?
- e.g. no syntax checking, typo checking in property keys, 
- e.g. how do you ensure that Polish localization `arb` file has all three hundred translation keys that are present in english `arb` file? What if I have 15 `arb` files to maintain

Perhaps, Localizely platform that you mentioned solves above issues? 

Anyhow, ideal workflow that I expected is that whenever a new `Intl.message` gets added to a dart localization file, it would automatically get added & merged to existing `arb` files. If I have to manually run a command to do this, that isn't a big deal for me.

This way, the only remaining work that is left is to replace default translation with other translations. 

This way, I can guarantee that all translation keys will always be present in all `arb` files, and they will be in sync with each other all the time, while the only difference would be their translation texts.

**Comment by aleksakrstic** [comment]
@cielo 
If you use such Flutter Intl plugin for Android Studio, you can select a string in your Dart code, press `Alt+Enter` and extract the string to all ARB files, not just to the main file. Using that practice, you can easily maintain the same keys in all ARB files.

That extraction functionality is not yet available for VS Code plugin, but I expect it soon since the plugin for Android Studio and VS Code are developed in parallel.

**Comment by guyzk** [comment]
Coming from Android, the Android Studio built in tool for handling translations is pretty awesome.
As it is laid out in a table you could instantly see which translations are missing, copy paste from an external spreadsheet etc...
I wish that it could be implemented for Flutter translations as well. It's a shame it is taking the Flutter team so long to find a good solution as a fundamental part of an enterprise (global) app is localisation.

**Comment by tapionx** [comment]
This is a real show stopper. Any internationalization toolkit should provide you with a way to delete obsolete strings and add new ones for each locale, and generate files ready to be sent to the translators. 
I don't see the need to reinvent the wheel and create a library without this basic features, when one could just adopt the GNU gettext standard.
👎
